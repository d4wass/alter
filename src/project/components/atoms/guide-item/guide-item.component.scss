@use '../../../../mixins';
@use '../../../../base';

.img-container{
    display: flex;
    justify-content: center;
    align-items: center;
    img{
        width: 200px;
        height: 200px;
    }
}

.wrapper{
    @include mixins.subtitle(25px);
    @include mixins.paragraph;
    border-radius: base.$main-border-rad;
    box-shadow: 0 4px 6px 0 hsla(0, 0%, 0%, 0.2);
    height: 100%;
    padding: 20px;

    h3{
        margin: 0
    }
}