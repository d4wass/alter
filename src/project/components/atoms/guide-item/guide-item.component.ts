import { Component, Input } from '@angular/core';

@Component({
    selector: 'app-guide-item',
    template: `
    <div class="wrapper">
      <div class="img-container">
        <img src="{{ graphic }}" *ngIf="graphic" data-test="image" />
      </div>
      <h3 data-test="title">{{ title }}</h3>
      <p data-test="content">{{ content }}</p>
    </div>
  `,
    styleUrls: ['./guide-item.component.scss'],
    standalone: false
})
export class GuideItemComponent {
  @Input() title!: string;
  @Input() content!: string;
  @Input() graphic?: string;
}
