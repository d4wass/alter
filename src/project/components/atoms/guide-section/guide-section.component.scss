@use '../../../../_mixins';

.wrapper{
    @include mixins.title(50px);
}

.guides-container {
    padding: 50px 0 70px 0;
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    grid-template-rows: 450px;
    grid-gap: 30px;
}
