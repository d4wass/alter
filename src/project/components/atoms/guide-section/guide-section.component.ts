import { Component } from '@angular/core';

const guidesItems = [
  {
    graphic: 'assets/host/graphic/car.svg',
    title: 'Getting started guide',
    content:
      'From getting to know the app to tweaking your settings to work for your schedule, get a comprehensive crash course in setting up and optimizing your listing.'
  },
  {
    graphic: 'assets/host/graphic/camera.svg',
    title: 'Photographic guide',
    content:
      'Better pics mean more clicks — see how to take strong shots of your car to make your car sparkle and stand out in search results.'
  },
  {
    graphic: 'assets/host/graphic/clean.svg',
    title: 'Cleaning & disnfection guide',
    content:
      'See how to clean and disinfect your car thoroughly and effectively to safeguard yourself and your guests, and comply with the Turo enhanced cleaning policy.'
  },
  {
    graphic: 'assets/host/graphic/contactless.svg',
    title: 'Contactless check-in guide',
    content:
      'Maintain physical distance and promote peace of mind with these tips and best practices on how to check in your guests remotely.'
  },
  {
    graphic: 'assets/host/graphic/protection.svg',
    title: 'How it works: Vehicle protection for Turo hosts',
    content:
      'Explore your protection plan options, get to know the terminology, get your questions answered, and pick the plan that’s best for you.'
  }
];

@Component({
  selector: 'app-guide-section',
  template: `
    <div class="wrapper">
      <h1>Guides to help you learn & earn</h1>
      <div class="guides-container">
        <app-guide-item
          *ngFor="let item of guidesItems"
          [title]="item.title"
          [content]="item.content"
          [graphic]="item.graphic"
        ></app-guide-item>
      </div>
    </div>
  `,
  styleUrls: ['./guide-section.component.scss']
})
export class GuideSectionComponent {
  guidesItems = guidesItems;
}
