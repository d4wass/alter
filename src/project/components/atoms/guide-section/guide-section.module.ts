import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { GuideSectionComponent } from './guide-section.component';
import { GuideItemComponent } from '../guide-item/guide-item.component';

@NgModule({
  declarations: [GuideSectionComponent, GuideItemComponent],
  imports: [CommonModule],
  exports: [GuideSectionComponent]
})
export class GuideSectionModule {}
