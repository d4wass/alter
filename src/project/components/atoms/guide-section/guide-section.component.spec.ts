import { createComponentFactory, Spectator } from '@ngneat/spectator';
import { MockComponent } from 'ng-mocks';
import { GuideItemComponent } from '../guide-item/guide-item.component';

import { GuideSectionComponent } from './guide-section.component';

describe('GuideSectionComponent', () => {
  let spectator: Spectator<GuideSectionComponent>;
  let component: GuideSectionComponent;

  const createComponent = createComponentFactory({
    component: GuideSectionComponent,
    declarations: [MockComponent(GuideItemComponent)]
  });

  beforeEach(() => {
    spectator = createComponent();
    component = spectator.component;
  });

  it('should match to snapshot', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
