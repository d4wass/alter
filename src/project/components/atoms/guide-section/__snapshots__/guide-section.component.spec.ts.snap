// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuideSectionComponent should match to snapshot 1`] = `
<app-guide-section
  guidesItems={[Function Array]}
>
  <div
    class="wrapper"
  >
    <h1>
      Guides to help you learn & earn
    </h1>
    <div
      class="guides-container"
    >
      <app-guide-item />
      <app-guide-item />
      <app-guide-item />
      <app-guide-item />
      <app-guide-item />
    </div>
  </div>
</app-guide-section>
`;
